src/client/method-default-target.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Contributors to the Eclipse BlueChi project
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+#include "method-default-target.h"
+#include "client.h"
+#include "usage.h"
+
+#include "libbluechi/common/opt.h"
+
+static int method_get_default_target_on(Client *client, char *node_name) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        char *result = NULL;
+        int r = 0;
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        BC_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        "GetDefaultTarget",
+                        &error,
+                        &message,
+                        "");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_read(message, "s", &result);
+        if (r < 0) {
+                fprintf(stderr, "Failed to parse response message: %s\n", strerror(-r));
+                return r;
+        }
+
+        printf("%s\n", result);
+
+        return r;
+}
+
+
+static int method_set_default_target_on(Client *client, char *node_name, char *target, char *force) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        int r = 0;
+        bool force_b = 1;
+        if (!strcmp(force, "false")) {
```

Comment:
Hmm, shouldn't the condition be reversed?

```
bool force_b = 1;
if (streq(force, "false")) {
         force_b = 0;
}
```

Reply:
Changed